[stable/cluster-autoscaler] Trim down PodSecurityPolicy

gypsydiver opened the pull request:
Signed-off-by: Fernando Mendoza <fermendozarcs@gmail.com>

<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

#### What this PR does / why we need it:
Current PSP is rather broad, this PR trims it down.

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

gypsydiver committed:
Trim down PodSecurityPolicy

Signed-off-by: Fernando Mendoza <fermendozarcs@gmail.com>

gypsydiver committed:
Merge branch 'master' into gypsydiver/cluster-autoscaler-trim-down-psp

gypsydiver committed:
Bump Chart Version

Signed-off-by: Fernando Mendoza <fermendozarcs@gmail.com>

yurrriq commented on stable/cluster-autoscaler/Chart.yaml:
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 icon: https://github.com/kubernetes/kubernetes/blob/master/logo/logo.png
 name: cluster-autoscaler
-version: 0.14.2
+version: 0.15.0
```
```suggestion
version: 2.0.0
```

Since this radically changes the PSP, and might not match people's expectations, I think a major version bump is in order, and _master_ is at `1.1.0` as I type this.

yurrriq commented on stable/cluster-autoscaler/templates/clusterrole.yaml:
```diff
@@ -104,11 +104,11 @@ rules:
 {{- if .Values.rbac.pspEnabled }}
   - apiGroups:
-    - extensions
+    - policy
```
Does this work on older versions of Kubernetes? Should we branch on version?

gypsydiver replied:
`extensions` would be right up to v1.13, v1.14 and on would "require" `policy`. It seems the API is kept backwards compatible for now and `extensions` could still be used up to v1.15. We could branch on API version, or just leave it at `extensions` for now.

yurrriq replied:
Let's maybe leave it as `extensions` for now. Your call.

yurrriq commented on stable/cluster-autoscaler/templates/podsecuritypolicy.yaml:
```diff
@@ -1,26 +1,42 @@
 {{- if .Values.rbac.pspEnabled }}
```
Will you please document the PSP in the README (I've just noticed it's entirely absent!), and make a brief note that this PR takes quite a departure from the previous versions?

yurrriq reviewed (changes requested):
Thanks for this! I've left a few requests and question inline.

gypsydiver committed:
Merge branch 'master' into gypsydiver/cluster-autoscaler-trim-down-psp

gypsydiver committed:
chart version major bump

Signed-off-by: Fernando Mendoza <fermendozarcs@gmail.com>

gypsydiver committed:
chart version major bump

Signed-off-by: Fernando Mendoza <fermendozarcs@gmail.com>

gypsydiver committed:
chart version major bump

Signed-off-by: Fernando Mendoza <fermendozarcs@gmail.com>

Commit:
[stable/prometheus-operator] Fix #15819 datasource config from k8s secrets (#15855)

* [stable/prometheus-operator] Fix #15819

Fix #15819 issue configuring datasources from kubernetes secrets

Changes in the stable/grafana charts allows for the recommended updating
and removal of datasources via kubernetes secrets. Older versions of
this chart do not allow this feature.

Signed-off-by: John Naulty <johnnaulty@bitgo.com>

* [stable/prometheus-operator] bump up minor version

This is adding feature, so minor version should be updaated, not patch
version.

Signed-off-by: John Naulty <johnnaulty@bitgo.com>

monotek committed:
changed mysql password usage to prevent warnings & used labels the helm 2.14 way (#15893)

Signed-off-by: André Bauer <andre.bauer@kiwigrid.com>

pierluigilenoci committed:
[stable/heapster] New version (#14815)

* New heapster version

Signed-off-by: Pierluigi Lenoci <pierluigi.lenoci@akelius.de>

* Update README.md

Signed-off-by: Pierluigi Lenoci <pierluigi.lenoci@akelius.de>

* Let's try in this way

Signed-off-by: Pierluigi Lenoci <pierluigi.lenoci@akelius.de>

DaemonDude23 committed:
Allow mounting extra volumes (#15860)

Signed-off-by: Andrew Aadland <aa23@protonmail.com>

Allows one to mount any number of volumes into the pod.

This is useful, for example, when sending email to an SMTP relay that
requires authentication (and thus the `smtp_auth_file` config parameter
to point to a path on the file system). An example for this is commented

isindir committed:
[stable/newrelic-infrastructure] fix NOTES.txt requirements checkup; flatten custom secret structure for better evaluation of the values (#15806)

Signed-off-by: Eriks Zelenka <isindir@users.sf.net>

naseemkullah committed:
bump version (#15865)

Signed-off-by: Naseem <naseemkullah@gmail.com>

tompizmor committed:
[stable/wordpress] Update dependencies for wordpress (#15889)

Signed-off-by: tompizmor <tompizmor@gmail.com>

YoussB committed:
stable/concourse: remove workerKeyPub from worker secrets (#15579)

- removed references to the worker public key from the worker deployment
- on the Concourse side, added more tests to make sure this is not used
anywhere (see: concourse/concourse#4133)

fixes: #13019

Signed-off-by: Zoe Tian <ztian@pivotal.io>
Co-authored-by: Bishoy Youssef <byoussef@pivotal.io>
Co-authored-by: Ciro S. Costa <cscosta@pivotal.io>

yurrriq commented on stable/cluster-autoscaler/Chart.yaml:
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 icon: https://github.com/kubernetes/kubernetes/blob/master/logo/logo.png
 name: cluster-autoscaler
-version: 1.1.0
+version: 2.0.0
```
Heads up this will likely need to change again, as https://github.com/helm/charts/pull/15695 is on its way to getting merged.

```suggestion
version: 3.0.0
```

gypsydiver replied:
Thanks for the heads up!

tompizmor committed:
[stable/phabricator] Update dependencies (#15871)

Signed-off-by: tompizmor <tompizmor@gmail.com>

tompizmor committed:
[stable/parse] Update dependencies (#15872)

Signed-off-by: tompizmor <tompizmor@gmail.com>

tompizmor committed:
[stable/redmine] Update dependencies (#15873)

Signed-off-by: tompizmor <tompizmor@gmail.com>

tompizmor committed:
[stable/odoo] Update dependencies (#15874)

Signed-off-by: tompizmor <tompizmor@gmail.com>

tompizmor committed:
[stable/drupal] Update dependencies for drupal (#15875)

Signed-off-by: tompizmor <tompizmor@gmail.com>

rifelpet committed:
[stable/cluster-autoscaler] Use labels recommended by Helm (#15695)

See https://helm.sh/docs/chart_best_practices/#standard-labels

Since the Deployment's selector is immutable, this is breaking change and will require a deletion and recreation, hence the major version bump.

Signed-off-by: Peter Rifel <pgrifel@gmail.com>

scuzhanglei committed:
Update README.md (#15868)

In fact,  default-backend-service is not required. See https://github.com/kubernetes/ingress-nginx/pull/3126

Signed-off-by: zhanglei <zhanglei@smartx.com>

tompizmor committed:
[stable/jasperreports] Update dependencies for jasperreports (#15876)

Signed-off-by: tompizmor <tompizmor@gmail.com>

tompizmor committed:
[stable/joomla] Update dependencies for joomla (#15877)

* [stable/joomla] Update dependencies for joomla

Signed-off-by: tompizmor <tompizmor@gmail.com>

* Specify mariadb version to use by Joomla! app

Signed-off-by: tompizmor <tompizmor@gmail.com>

tompizmor committed:
[stable/mediawiki] Update dependencies for mediawiki (#15878)

Signed-off-by: tompizmor <tompizmor@gmail.com>

wonderflow committed:
add helm hub china in README.md (#15004)

Signed-off-by: 天元 <jianbo.sjb@alibaba-inc.com>

tompizmor committed:
[stable/moodle] Update dependencies for moodle (#15879)

Signed-off-by: tompizmor <tompizmor@gmail.com>

tompizmor committed:
[stable/opencart] Update dependencies for opencart (#15880)

Signed-off-by: tompizmor <tompizmor@gmail.com>

tompizmor committed:
[stable/orangehrm] Update dependencies for orangehrm (#15881)

Signed-off-by: tompizmor <tompizmor@gmail.com>

tompizmor committed:
[stable/osclass] Update dependencies for osclass (#15882)

Signed-off-by: tompizmor <tompizmor@gmail.com>

tompizmor committed:
[stable/owncloud] Update dependencies for owncloud (#15883)

Signed-off-by: tompizmor <tompizmor@gmail.com>

tompizmor committed:
[stable/phpbb] Update dependencies for phpbb (#15884)

Signed-off-by: tompizmor <tompizmor@gmail.com>

tompizmor committed:
[stable/phpmyadmin] Update dependencies for phpmyadmin (#15885)

Signed-off-by: tompizmor <tompizmor@gmail.com>

tompizmor committed:
[stable/prestashop] Update dependencies for prestashop (#15886)

* [stable/prestashop] Update dependencies for prestashop

Signed-off-by: tompizmor <tompizmor@gmail.com>

* Specify mariadb version to use by Prestashop app

Signed-off-by: tompizmor <tompizmor@gmail.com>

tompizmor committed:
[stable/suitecrm] Update dependencies for suitecrm (#15887)

Signed-off-by: tompizmor <tompizmor@gmail.com>
